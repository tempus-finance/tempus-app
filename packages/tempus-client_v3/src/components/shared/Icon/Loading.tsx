import React, { FC } from 'react';
import { IconProps } from './index';
import withIcon from './withIcon';

const Loading: FC<IconProps> = ({ size = 16 }) => (
  <svg width={size} height={size} viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
    <g opacity="0.5">
      <path
        d="M7.99999 0C6.41774 0 4.87102 0.469191 3.55543 1.34824C2.23984 2.22729 1.21447 3.47672 0.608966 4.93852C0.00346627 6.40033 -0.15496 8.00886 0.153721 9.5607C0.462402 11.1125 1.22433 12.538 2.34314 13.6568C3.46196 14.7756 4.88742 15.5376 6.43927 15.8462C7.99111 16.1549 9.59964 15.9965 11.0614 15.391C12.5233 14.7855 13.7727 13.7601 14.6517 12.4445C15.5308 11.1289 16 9.58223 16 7.99998C16 6.94941 15.793 5.90912 15.391 4.93852C14.989 3.96792 14.3997 3.08601 13.6568 2.34314C12.914 1.60027 12.0321 1.011 11.0614 0.608962C10.0908 0.206926 9.05056 0 7.99999 0ZM7.99999 14.4C6.73419 14.4 5.49682 14.0246 4.44434 13.3214C3.39187 12.6181 2.57157 11.6186 2.08717 10.4492C1.60277 9.27971 1.47603 7.99288 1.72297 6.75141C1.96992 5.50993 2.57946 4.36956 3.47451 3.47451C4.36957 2.57945 5.50994 1.96992 6.75141 1.72297C7.99289 1.47603 9.27971 1.60277 10.4492 2.08717C11.6186 2.57157 12.6181 3.39187 13.3214 4.44434C14.0246 5.49681 14.4 6.73418 14.4 7.99998C14.4 9.69736 13.7257 11.3252 12.5255 12.5255C11.3252 13.7257 9.69737 14.4 7.99999 14.4Z"
        fill="url(#paint0_linear_168_2615)"
      />
      <path
        d="M7.99999 0C6.41774 0 4.87102 0.469191 3.55543 1.34824C2.23984 2.22729 1.21447 3.47672 0.608966 4.93852C0.00346627 6.40033 -0.15496 8.00886 0.153721 9.5607C0.462402 11.1125 1.22433 12.538 2.34314 13.6568C3.46196 14.7756 4.88742 15.5376 6.43927 15.8462C7.99111 16.1549 9.59964 15.9965 11.0614 15.391C12.5233 14.7855 13.7727 13.7601 14.6517 12.4445C15.5308 11.1289 16 9.58223 16 7.99998C16 6.94941 15.793 5.90912 15.391 4.93852C14.989 3.96792 14.3997 3.08601 13.6568 2.34314C12.914 1.60027 12.0321 1.011 11.0614 0.608962C10.0908 0.206926 9.05056 0 7.99999 0ZM7.99999 14.4C6.73419 14.4 5.49682 14.0246 4.44434 13.3214C3.39187 12.6181 2.57157 11.6186 2.08717 10.4492C1.60277 9.27971 1.47603 7.99288 1.72297 6.75141C1.96992 5.50993 2.57946 4.36956 3.47451 3.47451C4.36957 2.57945 5.50994 1.96992 6.75141 1.72297C7.99289 1.47603 9.27971 1.60277 10.4492 2.08717C11.6186 2.57157 12.6181 3.39187 13.3214 4.44434C14.0246 5.49681 14.4 6.73418 14.4 7.99998C14.4 9.69736 13.7257 11.3252 12.5255 12.5255C11.3252 13.7257 9.69737 14.4 7.99999 14.4Z"
        fill="url(#paint1_radial_168_2615)"
        fillOpacity="0.3"
      />
    </g>
    <path
      d="M14.4 7.99998H16C16 6.94941 15.7931 5.90912 15.391 4.93852C14.989 3.96792 14.3997 3.08601 13.6568 2.34314C12.914 1.60027 12.0321 1.011 11.0615 0.608962C10.0909 0.206926 9.05057 0 8 0V1.6C9.69738 1.6 11.3252 2.27428 12.5255 3.47451C13.7257 4.67474 14.4 6.3026 14.4 7.99998Z"
      fill="url(#paint2_linear_168_2615)"
    />
    <path
      d="M14.4 7.99998H16C16 6.94941 15.7931 5.90912 15.391 4.93852C14.989 3.96792 14.3997 3.08601 13.6568 2.34314C12.914 1.60027 12.0321 1.011 11.0615 0.608962C10.0909 0.206926 9.05057 0 8 0V1.6C9.69738 1.6 11.3252 2.27428 12.5255 3.47451C13.7257 4.67474 14.4 6.3026 14.4 7.99998Z"
      fill="url(#paint3_radial_168_2615)"
      fillOpacity="0.3"
    />
    <defs>
      <linearGradient
        id="paint0_linear_168_2615"
        x1="0.993967"
        y1="1.58721"
        x2="19.0889"
        y2="10.4375"
        gradientUnits="userSpaceOnUse"
      >
        <stop stopColor="#4B4B4B" />
        <stop offset="1" stopColor="#00060B" />
      </linearGradient>
      <radialGradient
        id="paint1_radial_168_2615"
        cx="0"
        cy="0"
        r="1"
        gradientUnits="userSpaceOnUse"
        gradientTransform="translate(13.5263 2.77244) rotate(115.648) scale(12.2286 10.7994)"
      >
        <stop stopColor="#00C2FF" />
        <stop offset="1" stopColor="#0057FF" stopOpacity="0.63" />
      </radialGradient>
      <linearGradient
        id="paint2_linear_168_2615"
        x1="8.49698"
        y1="0.793607"
        x2="17.5445"
        y2="5.21875"
        gradientUnits="userSpaceOnUse"
      >
        <stop stopColor="#4B4B4B" />
        <stop offset="1" stopColor="#00060B" />
      </linearGradient>
      <radialGradient
        id="paint3_radial_168_2615"
        cx="0"
        cy="0"
        r="1"
        gradientUnits="userSpaceOnUse"
        gradientTransform="translate(14.7632 1.38622) rotate(115.648) scale(6.11429 5.39971)"
      >
        <stop stopColor="#00C2FF" />
        <stop offset="1" stopColor="#0057FF" stopOpacity="0.63" />
      </radialGradient>
    </defs>
  </svg>
);

export default withIcon(Loading);
