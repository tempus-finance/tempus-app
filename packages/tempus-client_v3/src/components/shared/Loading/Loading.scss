.tc__loading {
  fill: none;
  stroke-width: 10px;
  stroke-linecap: square;
}

.tc__loading__bg {
  stroke: rgba(255, 255, 255, 0.5);
}

.tc__loading__animate {
  transform-origin: 50px 50px 0;
  stroke: rgba(255, 255, 255, 1);
  stroke-dasharray: 250;
  animation: circle-animation 4s linear infinite;
}

@keyframes circle-animation {
  0% {
    stroke-dashoffset: 40;
    transform: rotate(0deg);
  }
  50% {
    stroke-dashoffset: 150;
    transform: rotate(540deg);
  }
  100% {
    stroke-dashoffset: 40;
    transform: rotate(1080deg);
  }
}
