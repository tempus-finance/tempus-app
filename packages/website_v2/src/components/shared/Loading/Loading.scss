.tw__loading {
  fill: none;
  stroke-width: 10px;
  stroke-linecap: square;
}

.tw__loading__color-bg-default {
  stroke: rgba(255, 255, 255, 0.5);
}

.tw__loading__animate {
  transform-origin: 50px 50px 0;
  stroke-dasharray: 250;
  animation: circle-animation 4s linear infinite;
}

.tw__loading__color-animate-default {
  stroke: rgba(255, 255, 255, 1);
}

@keyframes circle-animation {
  0% {
    stroke-dashoffset: 40;
    transform: rotate(0deg);
  }
  50% {
    stroke-dashoffset: 150;
    transform: rotate(540deg);
  }
  100% {
    stroke-dashoffset: 40;
    transform: rotate(1080deg);
  }
}
