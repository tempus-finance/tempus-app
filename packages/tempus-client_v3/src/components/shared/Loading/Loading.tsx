import { FC } from 'react';

import './Loading.scss';

export type LoadingColor = 'default' | 'primary' | 'secondary';

export interface LoadingProps {
  size?: number;
  color?: LoadingColor;
}

const DEFAULT_SIZE = 20;
const DEFAULT_COLOR: LoadingColor = 'default';

const Loading: FC<LoadingProps> = ({ size = DEFAULT_SIZE, color = DEFAULT_COLOR }) => (
  <svg height={size} width={size} viewBox="0 0 100 100">
    <circle className={`tc__loading tc__loading__color-bg-${color}`} cx="50" cy="50" r="40" />
    <circle className={`tc__loading tc__loading__color-animate-${color} tc__loading__animate`} cx="50" cy="50" r="40" />
  </svg>
);

export default Loading;
