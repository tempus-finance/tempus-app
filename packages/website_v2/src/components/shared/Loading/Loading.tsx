import React, { FC } from 'react';
import './Loading.scss';

type LoadingVariant = 'light' | 'dark';

interface LoadingProps {
  variant?: LoadingVariant;
}

const Loading: FC<LoadingProps> = props => {
  const { variant = 'dark' } = props;

  return (
    <svg height="50" width="50" viewBox="0 0 100 100">
      <circle className={`tw__loading tw__loading__color-bg-${variant}`} cx="50" cy="50" r="40" />
      <circle
        className={`tw__loading tw__loading__color-animate-${variant} tw__loading__animate`}
        cx="50"
        cy="50"
        r="40"
      />
    </svg>
  );
};

export default React.memo(Loading);
