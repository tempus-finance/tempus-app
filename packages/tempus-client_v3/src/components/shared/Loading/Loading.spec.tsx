import { render } from '@testing-library/react';
import Loading, { LoadingColor, LoadingProps } from './Loading';

const subject = (props: LoadingProps) => render(<Loading {...props} />);

describe('Loading', () => {
  it('renders a loading', () => {
    const { container } = subject({});

    const svg = container.querySelector('svg');
    const loadingCircle = container.querySelector('circle');

    expect(loadingCircle).not.toBeNull();
    expect(loadingCircle).toHaveClass('tc__loading');

    expect(svg).not.toBeNull();
    expect(svg).toMatchSnapshot();
  });

  (['default', 'secondary'] as LoadingColor[]).forEach(color => {
    it(`renders a default-sized loading with ${color} color`, () => {
      const { container } = subject({ color });

      const svg = container.querySelector('svg');
      const loadingCircle = container.querySelector('circle');

      expect(loadingCircle).not.toBeNull();
      expect(loadingCircle).toHaveClass('tc__loading');
      expect(loadingCircle).toHaveClass(`tc__loading__color-bg-${color}`);

      expect(svg).not.toBeNull();
      expect(svg).toMatchSnapshot();
    });

    it(`renders a loading with ${color} color and size of 16 pixels`, () => {
      const { container } = subject({ size: 16, color });

      const svg = container.querySelector('svg');
      const loadingCircle = container.querySelector('circle');

      expect(loadingCircle).not.toBeNull();
      expect(loadingCircle).toHaveClass('tc__loading');
      expect(loadingCircle).toHaveClass(`tc__loading__color-bg-${color}`);

      expect(svg).not.toBeNull();
      expect(svg).toMatchSnapshot();
    });
  });
});
