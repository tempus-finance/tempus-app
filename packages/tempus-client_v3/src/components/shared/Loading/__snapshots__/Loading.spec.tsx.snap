// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders a default-sized loading with default color 1`] = `
<svg
  height="20"
  viewBox="0 0 100 100"
  width="20"
>
  <circle
    class="tc__loading tc__loading__color-bg-default"
    cx="50"
    cy="50"
    r="40"
  />
  <circle
    class="tc__loading tc__loading__color-animate-default tc__loading__animate"
    cx="50"
    cy="50"
    r="40"
  />
</svg>
`;

exports[`Icon renders a default-sized loading with secondary color 1`] = `
<svg
  height="20"
  viewBox="0 0 100 100"
  width="20"
>
  <circle
    class="tc__loading tc__loading__color-bg-secondary"
    cx="50"
    cy="50"
    r="40"
  />
  <circle
    class="tc__loading tc__loading__color-animate-secondary tc__loading__animate"
    cx="50"
    cy="50"
    r="40"
  />
</svg>
`;

exports[`Icon renders a loading 1`] = `
<svg
  height="20"
  viewBox="0 0 100 100"
  width="20"
>
  <circle
    class="tc__loading tc__loading__color-bg-default"
    cx="50"
    cy="50"
    r="40"
  />
  <circle
    class="tc__loading tc__loading__color-animate-default tc__loading__animate"
    cx="50"
    cy="50"
    r="40"
  />
</svg>
`;

exports[`Icon renders a loading with default color and size of 16 pixels 1`] = `
<svg
  height="16"
  viewBox="0 0 100 100"
  width="16"
>
  <circle
    class="tc__loading tc__loading__color-bg-default"
    cx="50"
    cy="50"
    r="40"
  />
  <circle
    class="tc__loading tc__loading__color-animate-default tc__loading__animate"
    cx="50"
    cy="50"
    r="40"
  />
</svg>
`;

exports[`Icon renders a loading with secondary color and size of 16 pixels 1`] = `
<svg
  height="16"
  viewBox="0 0 100 100"
  width="16"
>
  <circle
    class="tc__loading tc__loading__color-bg-secondary"
    cx="50"
    cy="50"
    r="40"
  />
  <circle
    class="tc__loading tc__loading__color-animate-secondary tc__loading__animate"
    cx="50"
    cy="50"
    r="40"
  />
</svg>
`;
